tsdb/engine/tsm1/engine.go

```diff
@@ -1558,10 +1559,14 @@ func (e *Engine) WriteSnapshot() error {
 
 	var started *time.Time
 
+	log, logEnd := logger.NewOperation(e.logger, "Cache snapshot", "cache.snapshot")
+
 	defer func() {
 		if started != nil {
-			e.Cache.UpdateCompactTime(time.Since(*started))
-			e.logger.Info(fmt.Sprintf("Snapshot for path %s written in %v", e.path, time.Since(*started)))
+			elapsed := time.Since(*started)
+			e.Cache.UpdateCompactTime(elapsed)
+			e.logger.Info(fmt.Sprintf("Snapshot for path %s written in %v", e.path, elapsed))
+			logEnd()
```

Comment:
To my knowledge, `NewOperation` logs a start event, but this only logs an end event if started is not nil. Can you move `logEnd` out of this block to ensure it is always called?

It looks like, from the design of the code, it is impossible for started to be nil so this might not be necessary. The structure is a bit confusing though.

Reply:
Yes, I thought so too – started is always set. Most of the changes were avoiding logic changes to be of the lowest risk / impact. I'll see if it can be improved safely.

Reply:
I have updated the code, since `started` was always set